### ISSUE TITLE
Building fails due tue incorrect build-configuration with downloadable CS Libraries

### ISSUE THREAD

**Issue by MaxClerkwell** [copperspice/kitchensink#7]
I am running Debian and downloaded the prebuilt 1.7.4 libraries from [your server](https://download.copperspice.com/copperspice/binary/cs-1.7/
)
````
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
````

```cmake version 3.18.4```

```ninja version 1.10.1```

```gcc version 10.2.1 20210110 (Debian 10.2.1-6)``` 

After cloning and running 

````bash
cmake -G "Ninja" --clean-first \
-DCMAKE_CXX_FLAGS="-DQT_NO_OPENSSL" \
-DCMAKE_PREFIX_PATH=/usr/local/lib/copperspice/lib/cmake/CopperSpice \
-DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=../deploy .. \
&& ninja install
````

the build breaks at 

````bash
[54/84] Building CXX object src/CMakeF...henSink.dir/camera/imagesettings.cpp.o
FAILED: src/CMakeFiles/KitchenSink.dir/camera/imagesettings.cpp.o 
/usr/bin/c++ -DBUILD_DATE=\"08/10/2022\" -DQT_NO_WEBKIT -Isrc -I../src -I../src/audio -I../src/camera -I../src/graphics -I../src/network -I../src/views -I../src/widgets -isystem /usr/local/lib/copperspice/include -isystem /usr/local/lib/copperspice/include/QtCore -isystem /usr/local/lib/copperspice/include/QtGui -isystem /usr/local/lib/copperspice/include/QtNetwork -isystem /usr/local/lib/copperspice/include/QtOpenGL -isystem /usr/local/lib/copperspice/include/QtScript -isystem /usr/local/lib/copperspice/include/QtSql -isystem /usr/local/lib/copperspice/include/QtSvg -isystem /usr/local/lib/copperspice/include/QtXml -isystem /usr/local/lib/copperspice/include/QtXmlPatterns -isystem /usr/include/gstreamer-1.0 -isystem /usr/include/glib-2.0 -isystem /usr/lib/x86_64-linux-gnu/glib-2.0/include -isystem /usr/local/lib/copperspice/include/QtMultimedia -DQT_NO_OPENSSL -O3 -DNDEBUG -std=gnu++17 -MD -MT src/CMakeFiles/KitchenSink.dir/camera/imagesettings.cpp.o -MF src/CMakeFiles/KitchenSink.dir/camera/imagesettings.cpp.o.d -o src/CMakeFiles/KitchenSink.dir/camera/imagesettings.cpp.o -c ../src/camera/imagesettings.cpp
In file included from ../src/camera/imagesettings.cpp:23:
../src/camera/imagesettings.h:38:37: error: expected ‘)’ before ‘*’ token
   38 |    ImageSettings(QCameraImageCapture *imageCapture, QWidget *parent = nullptr);
      |                 ~                   ^~
      |                                     )
../src/camera/imagesettings.h:58:4: error: ‘QCameraImageCapture’ does not name a type
   58 |    QCameraImageCapture *imagecapture;
      |    ^~~~~~~~~~~~~~~~~~~
../src/camera/imagesettings.cpp:30:1: error: no declaration matches ‘ImageSettings::ImageSettings(QCameraImageCapture*, QWidget*)’
   30 | ImageSettings::ImageSettings(QCameraImageCapture *imageCapture, QWidget *parent)
      | ^~~~~~~~~~~~~
In file included from ../src/camera/imagesettings.cpp:23:
../src/camera/imagesettings.h:33:7: note: candidates are: ‘ImageSettings::ImageSettings(const ImageSettings&)’
   33 | class ImageSettings : public QDialog
      |       ^~~~~~~~~~~~~
../src/camera/imagesettings.h:33:7: note:                 ‘ImageSettings::ImageSettings()’
../src/camera/imagesettings.h:33:7: note: ‘class ImageSettings’ defined here
../src/camera/imagesettings.cpp: In member function ‘QImageEncoderSettings ImageSettings::imageSettings() const’:
../src/camera/imagesettings.cpp:75:37: error: ‘imagecapture’ was not declared in this scope
   75 |    QImageEncoderSettings settings = imagecapture->encodingSettings();
      |                                     ^~~~~~~~~~~~
[56/84] Building CXX object src/CMakeFiles/KitchenSink.dir/camera/camera.cpp.o
FAILED: src/CMakeFiles/KitchenSink.dir/camera/camera.cpp.o 
/usr/bin/c++ -DBUILD_DATE=\"08/10/2022\" -DQT_NO_WEBKIT -Isrc -I../src -I../src/audio -I../src/camera -I../src/graphics -I../src/network -I../src/views -I../src/widgets -isystem /usr/local/lib/copperspice/include -isystem /usr/local/lib/copperspice/include/QtCore -isystem /usr/local/lib/copperspice/include/QtGui -isystem /usr/local/lib/copperspice/include/QtNetwork -isystem /usr/local/lib/copperspice/include/QtOpenGL -isystem /usr/local/lib/copperspice/include/QtScript -isystem /usr/local/lib/copperspice/include/QtSql -isystem /usr/local/lib/copperspice/include/QtSvg -isystem /usr/local/lib/copperspice/include/QtXml -isystem /usr/local/lib/copperspice/include/QtXmlPatterns -isystem /usr/include/gstreamer-1.0 -isystem /usr/include/glib-2.0 -isystem /usr/lib/x86_64-linux-gnu/glib-2.0/include -isystem /usr/local/lib/copperspice/include/QtMultimedia -DQT_NO_OPENSSL -O3 -DNDEBUG -std=gnu++17 -MD -MT src/CMakeFiles/KitchenSink.dir/camera/camera.cpp.o -MF src/CMakeFiles/KitchenSink.dir/camera/camera.cpp.o.d -o src/CMakeFiles/KitchenSink.dir/camera/camera.cpp.o -c ../src/camera/camera.cpp
In file included from ../src/camera/camera.cpp:24:
../src/camera/imagesettings.h:54:28: error: ‘QComboBox’ does not name a type
   54 |    QVariant boxValue(const QComboBox *box) const;
      |                            ^~~~~~~~~
../src/camera/imagesettings.h:55:28: error: ‘QComboBox’ has not been declared
   55 |    void selectComboBoxItem(QComboBox *box, const QVariant &value);
      |                            ^~~~~~~~~
[59/84] Building CXX object src/CMakeF...Sink.dir/graphics/animated_tiles.cpp.o
ninja: build stopped: subcommand failed.
````

I assume, that an additional parameter is needed in the cmake invocation. Unfortunately this is not in the README.md.

I'd love to know the missing parameter :]

**Comment by bgeller** [comment]
I do not believe this is a CMake issue. Someone on the team will investigate.  

Barbara

**Comment by agserm** [comment]
Two forward declarations were accidentally removed. This issue has been resolved in commit 3ef3a4c. This commit restores the missing declaration.
